### ISSUE TITLE
Can't buy land or construction rights when in "No money" 

### ISSUE THREAD

**Issue by TheClaudeQC** [OpenRCT2/OpenRCT2#20988]
### Operating System

Windows 11, 64-Bits

### OpenRCT2 build

OpenRCT2, v0.4.6-155-g18dec7d (18dec7d on develop) provided by GitHub

### Base game

RollerCoaster Tycoon 2

### Area(s) with this issue?

The bug also exists in the original game (RCT2/RCTC)

### Describe the issue

You build a nice scenario whit buyable lands and/or right constructions and notice that it doesn't work in the scenario just because of the "no money" option on

### Steps to reproduce

1. Make any quick scenario
2. Put some land or Construction rights to buy
3. Put it in "No money"
4. Save scenario
5. Open said scenario
6. Try to buy land... oh wait! You can't!!!

### Attachments

![Test of a bug 2023-11-18 16-51-37](https://github.com/OpenRCT2/OpenRCT2/assets/83843734/a04d16d3-d590-43db-a1e3-da1b0603e904)
state of the game in the scenario play

![Test of a bug 2023-11-18 16-51-45](https://github.com/OpenRCT2/OpenRCT2/assets/83843734/d13a9186-904c-401e-8c42-bd1ac29f08bd)
State of the game when it was in the editor or you activated the "sandbox mode" cheat.

**Comment by HtotheTML** [comment]
Given that purchasable land would be free in a no-money scenario, the ability to buy it is a bit of a moot point.

I will say though that the land purchase window should still be accessible, as it is impossible to identify where construction rights are otherwise.

**Comment by ocalhoun6** [comment]
Yeah... The whole concept of purchasable land in a no money scenario is kind of silly.

Because wouldn't your first step just be to 'buy' all available land? There's no downside to that. So the scenario designer might as well just mark all that land as 'already owned' rather than purchasable. 

But yeah -- there does need to be some way to show where construction rights are in a no money scenario.
